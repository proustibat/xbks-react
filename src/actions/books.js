// SET_BOOKS
export const setBooks = books => ( {
    type: 'SET_BOOKS',
    books
} );
