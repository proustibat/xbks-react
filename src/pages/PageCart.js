import React from 'react';
import Cart from '../components/cart/Cart';

const PageCart = () => (
    <Cart />
);

export default PageCart;
