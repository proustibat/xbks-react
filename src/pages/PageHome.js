import React from 'react';
import BookList from '../components/BookList';

const PageHome = () => (
    <BookList />
);

export default PageHome;
