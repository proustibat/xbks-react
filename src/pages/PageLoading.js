import React from "react";

const PageLoading = () => (
  <div className="loader">
      <img
          className="loader__image"
          src="/images/loader.gif"
          alt="loader animation"
      />
  </div>
);

export default PageLoading;
