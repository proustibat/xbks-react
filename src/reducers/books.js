const booksReducerDefaultState = [];
export default ( state = booksReducerDefaultState, action ) => {
    switch ( action.type ) {
        case 'SET_BOOKS':
            return action.books;
        default:
            return state;
    }
};
