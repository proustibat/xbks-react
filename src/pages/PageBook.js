import React from 'react';
import { connect } from 'react-redux';
import BookCard from '../components/BookCard';
import { addBook } from "../actions/cart";
import { startSetOffers } from "../actions/offers";

export class PageBook extends React.Component {

    onAdd = () => {
        this.props.addToCart( this.props.book );
        this.props.startSetOffers();
        this.props.history.push( '/' );
    };

    render() {
        return (
            <div className="content-container flex-container-col">
                <BookCard
                    book = { this.props.book }
                />
                <button className="button button--secondary" onClick = { this.onAdd }>Buy this book</button>
            </div>
        );
    }
}

export const mapStateToProps = ( state, props ) => ( {
    book: state.books.find( book => book.isbn === props.match.params.isbn )
} );

export const mapDispatchToProps = dispatch => ( {
    addToCart: book => dispatch( addBook( book ) ),
    startSetOffers: () => dispatch ( startSetOffers() )
} );


export default connect( mapStateToProps, mapDispatchToProps )( PageBook );

