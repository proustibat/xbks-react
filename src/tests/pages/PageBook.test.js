import React from 'react';
import { shallow } from 'enzyme';
import { PageBook } from '../../pages/PageBook';
import books from '../fixtures/books';

test( 'Should render book page correctly', () => {
    const wrapper = shallow( <PageBook /> );
    expect( wrapper ).toMatchSnapshot();
} );

test( 'Should call cart and offers actions and history function when clicking on add button', () => {
    const addToCart = jest.fn();
    const startSetOffers = jest.fn();
    const history = { push: jest.fn() };
    const wrapper = shallow( <PageBook
        addToCart={ addToCart }
        startSetOffers={ startSetOffers }
        history={ history }
        book={ books[ 0 ] }
    /> );

    wrapper.find( 'button' ).simulate( 'click' );

    expect( history.push ).toHaveBeenLastCalledWith( '/' );
    expect( addToCart ).toHaveBeenLastCalledWith( books[ 0 ] );
    expect( startSetOffers ).toHaveBeenCalled();
} );
