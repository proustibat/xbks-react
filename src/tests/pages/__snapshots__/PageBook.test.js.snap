// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render book page correctly 1`] = `
<div
  className="content-container flex-container-col"
>
  <BookCard />
  <button
    className="button button--secondary"
    onClick={[Function]}
  >
    Buy this book
  </button>
</div>
`;
